tests/functional/replication/probe.spec.js

```diff
@@ -0,0 +1,72 @@
+const assert = require('assert');
+const sinon = require('sinon');
+const { startProbeServer } =
+    require('../../../extensions/replication/queueProcessor/Probe');
+const { DEFAULT_LIVE_ROUTE } = require('arsenal').network.probe.ProbeServer;
+const http = require('http');
+
+/**
+ * @returns {Object} Mock queue processor
+*/
+function mockQueueProcessor() {
+    return {
+        handleLiveness: sinon.stub(),
+    };
+}
+
+describe('Probe server', () => {
+    afterEach(() => {
+        // reset any possible env var set
+        delete process.env.CRR_METRICS_PROBE;
+    });
+
+    it('is not created when disabled', done => {
+        process.env.CRR_METRICS_PROBE = 'false';
+        const mockQp = mockQueueProcessor();
+        const config = {
+            bindAddress: 'localhost',
+            port: 52555,
+        };
+        startProbeServer(mockQp, config, probeServer => {
+            assert.strictEqual(probeServer, undefined);
+            done();
+        });
+    });
+
+    it('is not created with no config', done => {
+        process.env.CRR_METRICS_PROBE = 'true';
+        const mockQp = mockQueueProcessor();
+        const config = undefined;
+        startProbeServer(mockQp, config, probeServer => {
+            assert.strictEqual(probeServer, undefined);
+            done();
+        });
+    });
+
+    it('creates probe server for liveness', done => {
+        process.env.CRR_METRICS_PROBE = 'true';
+        const mockQp = mockQueueProcessor();
+        // return sample error for liveness
+        mockQp.handleLiveness.returns('error msg');
+        const config = {
+            bindAddress: 'localhost',
+            port: 52555,
+        };
+        startProbeServer(mockQp, config, probeServer => {
+            probeServer.onStop(done);
+            http.get(`http://localhost:52555${DEFAULT_LIVE_ROUTE}`, res => {
+                assert.strictEqual(500, res.statusCode);
```

Comment:
This should be `actual, expected` for the assertion check, I think its the other way around here.

Reply:
Is there any reason its 500 and not 200?

Reply:
The liveness check returns an error so the status code is 500, the test confirms the server is up and running and correctly handles the liveness response. The handle liveness is more thoroughly tested in unit tests.